pkgs/build-support/build-fhs-userenv/default.nix

```diff
@@ -0,0 +1,37 @@
+{ writeTextFile, stdenv, ruby } : { env, runScript } :
+
+let
+  name = env.pname;
+
+  # Sandboxing script
+  chroot-user = writeTextFile {
+    name = "chroot-user";
+    executable = true;
+    destination = "/bin/chroot-user";
+    text = ''
+      #! ${ruby}/bin/ruby
+      ${builtins.readFile ./chroot-user.rb}
```

Comment:
Well I was thinking that it looked similar to
https://www.mail-archive.com/nix-dev@lists.science.uu.nl/msg14656.html
where cacert gets built at eval time but must say I don't fully understand
how that happens. Maybe the problem there was the reference to pkgs.cacert.

On Thu Feb 05 2015 at 9:57:30 PM Nikolay Amiantov <notifications@github.com>
wrote:

> In pkgs/build-support/build-fhs-userenv/default.nix
> <https://github.com/NixOS/nixpkgs/pull/6177#discussion_r24198705>:
>
> > @@ -0,0 +1,37 @@
> > +{ writeTextFile, stdenv, ruby } : { env, runScript } :
> > +
> > +let
> > +  name = env.pname;
> > +
> > +  # Sandboxing script
> > +  chroot-user = writeTextFile {
> > +    name = "chroot-user";
> > +    executable = true;
> > +    destination = "/bin/chroot-user";
> > +    text = ''
> > +      #! ${ruby}/bin/ruby
> > +      ${builtins.readFile ./chroot-user.rb}
>
> Hm, I thought it would, too, but this file lies in nixpkgs and I supposed
> that we rely on it being unchanged while evaluating (otherwise lazy imports
> would be bad, too). Or have I completely misunderstood the issue?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/NixOS/nixpkgs/pull/6177/files#r24198705>.
>